lib/ansible/modules/windows/setup.ps1

```diff
@@ -358,8 +358,8 @@ if($gather_subset.Contains('processor')) {
         ansible_processor = $cpu_list
         ansible_processor_cores = $win32_cpu.NumberOfCores
         ansible_processor_count = $win32_cs.NumberOfProcessors
-        ansible_processor_threads_per_core = ($win32_cpu.NumberOfLogicalProcessors / $win32_cs.NumberOfProcessors / $win32_cpu.NumberOfCores)
-        ansible_processor_vcpus = ($win32_cpu.NumberOfLogicalProcessors / $win32_cs.NumberOfProcessors)
+        ansible_processor_threads_per_core = ($win32_cpu.NumberOfLogicalProcessors / $win32_cpu.NumberofCores)
+        ansible_processor_vcpus = ($win32_cpu.NumberOfLogicalProcessors * $win32_cs.NumberOfProcessors)
```

Comment:
This seems like it could simply be `$win32_cs.NumberOfLogicalProcessors`. This value already has the number of cores including hyperthreading so we don't have to do the calculations ourselves.

Reply:
$win32_cs.NumberOfLogicalProcessors is only equal to the number of cores plus hyper-threading on a single CPU system.  To get a true count you need to multiply logical processors via number of processors.

Reply:
That isn't what I am seeing on my tests, I spun up a 2016 instance on KVM in Fedora and have set the following CPU configuration

![image](https://user-images.githubusercontent.com/8462645/63658353-4fb5c580-c7ed-11e9-8a30-12cb024f3ee0.png)

Bascially I've set the VM to have 8 CPU sockets, 4 cores per socket, and finally 2 threads per core. When running this is what task manager shows

![image](https://user-images.githubusercontent.com/8462645/63658402-973c5180-c7ed-11e9-835e-7b930fcc2323.png)

Based on this configuration I would have 64 logical processors (8 * 4 * 2) and that is what it is showing me. The `NumberOfLogicalProcessors` is `64` and `NumberOfProcessors` is `8`

![image](https://user-images.githubusercontent.com/8462645/63658476-2cd7e100-c7ee-11e9-9b78-b42f923ba3fd.png)

This and the documentation tells me that `$win32_cs.NumberOfLogicalProcessors` is the true count of logical processors even with multiple CPUs.